### DISCUSSION TITLE
UseEffect are not working

### DISCUSSION THREAD

**Discussion by DennisWarchocki** [neuefische/questions#219] [Web] [answered]
Hello guys, 

I have the following problem:
I'm building a Timer function, and right now I have the Problem the Numbers within the Timer arent changing and just shown as static numbers. I cant understand why UseEffect not doing the work.

How my Counter looks:
![Bildschirmfoto 2022-10-05 um 20 24 40](https://user-images.githubusercontent.com/109968361/194134340-defaa544-6b2e-489f-ada5-34f186cafb72.png)

The Numbers should start at a given value and then count down to zero. 
My Numbers are just static and nothing is happen. I have that feeling I messed something up about the structure from my files but Im not sure. 

Would you be so kind and take a look at the issue? 

My branch:
https://github.com/DennisWarchocki/capstone-project/tree/timer

I get this displayed:

 
<img width="1027" alt="Bildschirmfoto 2022-10-07 um 10 04 34" src="https://user-images.githubusercontent.com/109968361/194504326-397544fd-37de-4e26-b5d6-308ca621b82a.png">

**Comment by manusmd** [comment]
Is this the latest code? I just see commented out code in the timer page

**Reply by DennisWarchocki** [reply]
> Is this the latest code? I just see commented out code in the timer page

I forgot to push something to GitHub, could you try it again now?

**Comment by mbosselmann** [comment, marked as answer]
Dennis and I worked on a solution together to make it work. We wrote a new `getTime` function and adjusted the callback function of the `useEffect`:

```
	const deadline = 'October, 8, 2022';

	function getTime() {
		const time = Date.parse(deadline) - Date.now();
		if (time > 0) {
			setTimerDays(Math.floor(time / (24 * 60 * 60 * 1000)));
			setTimerHours(Math.floor((time / (1000 * 60 * 60)) % 24));
			setTimerMinutes(Math.floor((time / 1000 / 60) % 60));
			setTimerSeconds(Math.floor((time / 1000) % 60));
		} else {
			setTimerDays(0);
			setTimerHours(0);
			setTimerMinutes(0);
			setTimerSeconds(0);
		}
	}

	useEffect(() => {
		const interval = setInterval(() => getTime(deadline), 1000);
		return () => clearInterval(interval);
	}, []);
```
Dennis now needs to customize the timer functionality to his liking so that it makes sense to the user. But the original problem is solved.
